src/state/image-dataset/csv-dataset/index.ts

```diff
@@ -252,22 +308,100 @@ class CsvRequest implements ImageDataset {
         // TODO: Feature defs can include units. Should we strip that from the feature column name?
     }
 
+    /**
+     * Assigns a default group-by feature key for the dataset. Datasets must have a
+     * discrete group-by feature or CFE will crash.
+     *
+     * Key is chosen in the following order:
+     * 1. Default BFF group by key ("Cell Line") if it exists
+     * 2. First discrete feature key if it exists
+     * 3. A default bin feature if no discrete feature exists
+     */
+    private assignDefaultGroupByFeatureKey(csvData: Record<string, string>[]): void {
+        // Check if the BFF-specific default group-by feature exists.
+        const firstRow = csvData[0];
+        if (firstRow && firstRow[BFF_DEFAULT_GROUP_BY_KEY] !== undefined) {
+            this.defaultGroupByFeatureKey = BFF_DEFAULT_GROUP_BY_KEY;
+            return;
+        }
+
+        // If not, assign the first discrete feature as the default group-by feature if it exists.
+        const firstDiscreteFeature = Array.from(this.featureInfo.values()).find(
+            (info) => info.type === FeatureType.DISCRETE
+        );
+        if (firstDiscreteFeature) {
+            this.defaultGroupByFeatureKey = firstDiscreteFeature.def.key;
+            return;
+        }
+
+        // If no discrete feature is found, assign a default group-by feature.
+        const options: Record<string, MeasuredFeaturesOption> = {
+            "0": {
+                color: DEFAULT_COLORS[0],
+                name: "Default",
+                key: "0",
+                count: csvData.length,
+            },
+        };
+        this.featureInfo.set(DEFAULT_GROUPBY_NONE, {
+            type: FeatureType.DISCRETE,
+            def: {
+                discrete: true,
+                displayName: "(None)",
+                description: "(None)",
+                key: DEFAULT_GROUPBY_NONE,
+                tooltip: "(None)",
+                options,
+            },
+            data: new Array(csvData.length).fill(0),
+        });
+        this.defaultGroupByFeatureKey = DEFAULT_GROUPBY_NONE;
+    }
+
+    private remapBffKeys = (row: Record<string, string>): void => {
```

Comment:
nit: separate out a helper function for checking undefined keys to reduce repetitive `if` statements?
```
if (row[A_KEY] === undefined && row[ANOTHER_KEY] !== undefined) {
row[A_KEY] = row[ANOTHER_KEY];
}
```

Reply:
Done!